packages/@aws-cdk/aws-codedeploy/lib/server/deployment-group.ts

```diff
@@ -269,7 +269,7 @@ export class ServerDeploymentGroup extends ServerDeploymentGroupBase {
       physicalName: props.deploymentGroupName,
     });
 
-    this.application = props.application || new ServerApplication(this, 'Application');
+    this.application = props.application || new ServerApplication(this, this.physicalName);
```

Comment:
This is not the right fix for this issue. The fix is that if the parent `DeploymentGroup` has a `PhysicalName.GENERATE_IF_NEEDED`, so should the `ServerApplication`.

The poster suggested as much under [Expected Result](https://github.com/aws/aws-cdk/issues/13337).

Also the title of this PR doesn't cover the change it's making. The title as written makes it sound as if no Application was created before, whereas now there is. This is not true though, an Application was already being created before. 

We generally also write the PR title in terms of things the user would care about, and specifically for fixes it describes the bug that is now fixed (users will see these titles in the CHANGELOG, so think about what would make sense to them when viewed there).

Reply:
@rix0rrr  Thanks for the review. I made the changes. Please have a look